packages/app/plugin/__tests__/iosPlugin.test.ts

```diff
@@ -0,0 +1,14 @@
+import fs from 'fs/promises';
+import path from 'path';
+
+import { modifyObjcAppDelegate } from '../src/ios/appDelegate';
+
+describe('Config Plugin iOS Tests', function () {
+  it('tests changes made to AppDelegate.m', async function () {
+    const appDelegate = await fs.readFile(path.join(__dirname, './fixtures/AppDelegate.m'), {
+      encoding: 'utf8',
+    });
+    const result = modifyObjcAppDelegate(appDelegate);
+    expect(result).toMatchSnapshot();
```

Comment:
We don't use snapshots anywhere else in our testing infrastructure, so it may be that the next person that has to figure out why this failed may have no experience with them (like me)

Could you leave a comment here with the steps you used to construct the test? It may seem obvious, but something like "fetch current template AppDelegate from (?), put it ?, do this thing to it and put it here" etc when I'm trying to keep this thing running in spare time a year from now and it's e.g. a swift file in the template now so it all breaks

Reply:
Yeah, I'll leave some doc comments there.

Basically, what the tests do is loading `build.gradle` or `AppDelegate.m` test fixtures (they're taken from [bare minimum Expo template](https://github.com/expo/expo/tree/master/templates/expo-template-bare-minimum)). Then it applies the plugin changes on them and compares changed files with snapshots. If they're equal - the tests pass.
If you intentionally change anything in plugin code (e.g. gradle dependency versions), you have to update the snapshots by running `yarn tests:jest -u` (or interactively from _watch_ mode by pressing the `i` key). Anyway, I'll make a small `README` in the `__tests__` directory.